app/models/katello/concerns/content_facet_host_extensions.rb

```diff
@@ -45,8 +45,14 @@ module ContentFacetHostExtensions
         accepts_nested_attributes_for(
           :content_facet,
           self.nested_attributes_options[:content_facet].merge(
-            :reject_if => proc { |attributes| attributes['content_view_id'].blank? && attributes['lifecycle_environment_id'].blank? })
+            :reject_if => :content_facet_creation_rejectable?)
         )
+
+        def content_facet_creation_rejectable?(attributes)
+          self.content_facet.blank? &&
+           attributes['content_view_id'].blank? &&
```

Comment:
Also, `attributes['content_source_id'].blank?`

Reply:
I don't think it does, the purpose of this is to reject content_facet_attributes if the content_view_id or lifecycle_environment_id are blank.  These are required attributes. 

If this was not here, if you tried to create a new host in the UI, but did not select content view and lifecycle environment, it would attempt to create a content facet (sending an empty hash for content_facet_attributes), and would fail because those two attributes are required on a content facet.  This allows the host to be created without a content facet in this case.

Reply:
Right now, you can set the content source on a host that doesn't have a CV or LE in the UI, and when you save it, the content source isn't actually set (and no content facet is created) and no error is given to the user. This is why I think we need an additional validation for content_source, whether here or somewhere else.

Reply:
Ah, good point @akofink is there an open issue on this?  

@parthaa i would recommend changing this to say reject if "content_view_id is blank or lifecycle_environment_id is blank or content_source_id is NOT blank", and then we could just rely on the existing validation to require content_view and lifecycle_environment

Reply:
This one is related to this lack of validation which is causing confusion in the UI and API http://projects.theforeman.org/issues/17644